src/main/kotlin/com/sourcegraph/cody/edit/EditShowDiffAction.kt

```diff
@@ -0,0 +1,65 @@
+package com.sourcegraph.cody.edit
+
+import com.intellij.diff.DiffContentFactory
+import com.intellij.diff.DiffRequestFactory
+import com.intellij.diff.actions.BlankDiffWindowUtil.createBlankDiffRequestChain
+import com.intellij.diff.actions.CompareFileWithEditorAction
+import com.intellij.diff.chains.DiffRequestChain
+import com.intellij.diff.util.DiffUserDataKeys
+import com.intellij.openapi.actionSystem.AnActionEvent
+import com.intellij.openapi.actionSystem.DataKey
+import com.intellij.openapi.editor.Document
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.editor.EditorFactory
+import com.intellij.openapi.fileEditor.FileDocumentManager
+import com.sourcegraph.cody.agent.protocol.Range
+
+class EditShowDiffAction : CompareFileWithEditorAction() {
+
+  override fun isAvailable(e: AnActionEvent): Boolean {
+    e.dataContext.getData(DOCUMENT_BEFORE_DATA_KEY) ?: return false
+    e.dataContext.getData(SELECTION_RANGE_DATA_KEY) ?: return false
+    return true
+  }
+
+  override fun getDiffRequestChain(e: AnActionEvent): DiffRequestChain {
+    val project = e.project
+    val selectionRange = e.dataContext.getData(SELECTION_RANGE_DATA_KEY)!!
```

Comment:
I don't think this is right. You should write a more detailed test plan, which includes editing (increasing and decreasting the line and character count) before, in, (and after, but that will probably work) the range:

- When the fixup is started, but before it is applied
- After the fixup is applied

And then showing the diff.

FixupTask on the extension side has a "live" range which tracks edits. But it appears that the JetBrains side is caching this range.

Reply:
I think it's working as expected now. There is a problem with changes when the fixup is started but not yet finished ("Cody is working..."). I does not work in that case but that case does not work for us at all yet.